neo4j/internal/bolt/bolt3_test.go

```diff
@@ -512,4 +513,39 @@ func TestBolt3(ot *testing.T) {
 		AssertNil(t, sum)
 		AssertError(t, err)
 	})
+
+	outer.Run("Updates idle date on every new message", func(t *testing.T) {
```

Comment:
```suggestion
	outer.Run("Updates idle date on SUCCESS after RESET", func(t *testing.T) {
```

It's not really testing every message.

Reply:
true, but the test implementation shows this, while the intention and the implementation shows it's done on every message

Reply:
ideally, the test would be more general and indeed test any message but that would be cumbersome to set up.
I'd like to keep the intention in the test name. Thoughts?

Reply:
Is there no easy way of setup a scenario for all messages? Like using the two functions per scenario, one to setup the mocks for the sever and other to call the method, this way you could use a parameterized test with pair of function for each message you wan't to check.

Reply:
updated with parameterizated test cases, covering almost every message